core/src/replay_stage.rs

```diff
@@ -613,6 +616,14 @@ impl ReplayStage {
                                     i64
                                 ),
                             );
+
+                            if my_pubkey != cluster_info.id() {
```

Comment:
ReplayStage id change needs to be timed with a fork reset, to ensure PoH recorder will be reset the include the next leader slot for the new id.  This effectively means that an id reset while the validator is leader will be postponed until their time as leader is up